cms/djangoapps/contentstore/views/preview.py

```diff
@@ -194,6 +196,17 @@ def _preview_module_system(request, descriptor, field_data):
         # stick the license wrapper in front
         wrappers.insert(0, partial(wrap_with_license, mako_service=mako_service))
 
+    preview_anonymous_user_id = 'student'
+    if individualize_anonymous_user_id(course_id):
+        # There are blocks (capa, html, and video) where we do not want to scope
+        # the anonymous_user_id to specific courses. These are captured in the
+        # block attribute 'requires_per_student_anonymous_id'. Please note,
+        # the course_id field in AnynomousUserID model is blank if value is None.
+        if getattr(descriptor, 'requires_per_student_anonymous_id', False):
+            preview_anonymous_user_id = anonymous_id_for_user(request.user, None)
+        else:
+            preview_anonymous_user_id = anonymous_id_for_user(request.user, course_id)
+
```

Comment:
This is working great @schenedx , exactly as expected.

But to make sure your change doesn't get broken somewhere down the line, we need to update the unit tests to test these different cases. Here's what I'd recommend: https://github.com/openedx/edx-platform/compare/schen/MST-1455...open-craft:jill/schen/MST-1455

Reply:
Great. I made [a PR](https://github.com/openedx/edx-platform/pull/30400) on the unit tests you wrote above and merged it. Once this PR is going green, I plan to merge as well.